awscli/customizations/history/show.py

```diff
@@ -0,0 +1,454 @@
+    def __init__(self, popen=None):
+        self._popen = popen
+        if popen is None:
+            self._popen = subprocess.Popen
+
+    def get_output_stream(self, stream_type):
+        """Get an output stream to write to
+
+        The value is wrapped in a context manager so make sure to use
+        a with statement.
+
+        :type stream_type: string
+        :param stream_type: The name of the stream to get. Valid values
+             consist of pager and stdout.
+        """
+        stream_getter = getattr(self, '_get_' + stream_type + '_stream', None)
+        if stream_getter is None:
+            raise ValueError(
+                'Stream type of %s is not supported' % stream_type)
+        return stream_getter()
+
+    @contextlib.contextmanager
+    def _get_pager_stream(self):
+        pager_cmd, kwargs = self._get_process_pager_with_kwargs()
+        process = self._popen(pager_cmd, **kwargs)
+        yield process.stdin
+        process.stdin.close()
+        process.wait()
+
+    @contextlib.contextmanager
+    def _get_stdout_stream(self):
+        yield get_binary_stdout()
+
+    def _get_process_pager_with_kwargs(self):
+        pager_cmd, kwargs = get_popen_pager_cmd_with_kwargs(
+            os.environ.get('PAGER'))
+        kwargs['stdin'] = subprocess.PIPE
+        return pager_cmd, kwargs
+
+
+class ShowCommand(BasicCommand):
+    NAME = 'show'
+    DESCRIPTION = (
+        'Shows the various events related to running a specific CLI command. '
+        'If this command is ran without any positional arguments, it will '
+        'display the events for the last CLI command ran.'
+    )
+    FORMATTERS = {
+        'detailed': DetailedFormatter
+    }
+    ARG_TABLE = [
+        {'name': 'command_id', 'nargs': '?', 'default': 'latest',
+         'positional_arg': True,
+         'help_text': (
+             'The ID of the CLI command to show. If this positional argument '
+             'is omitted, it will show the last the CLI command ran.')},
+        {'name': 'include', 'nargs': '+',
+         'help_text': (
+             'Specifies which events to **only** include when showing the '
+             'CLI command. This argument is mutually exclusive with '
+             '``--exclude``.')},
+        {'name': 'exclude', 'nargs': '+',
+         'help_text': (
+             'Specifies which events to exclude when showing the '
+             'CLI command. This argument is mutually exclusive with '
+             '``--include``.')},
+        {'name': 'format', 'choices': FORMATTERS.keys(),
+         'default': 'detailed', 'help_text': (
+            'Specifies which format to use in showing the events for '
+            'the specified CLI command.')}
+    ]
+
+    def __init__(self, session, db_reader=None, output_stream_factory=None):
+        super(ShowCommand, self).__init__(session)
+        self._db_reader = db_reader
+        if db_reader is None:
+            connection = DatabaseConnection(
```

Comment:
This logic was removed from the DatabaseConnection class.

Reply:
Yep I added that logic back in a recent commit